deeposlandia/datasets/mapillary.py

```diff
@@ -0,0 +1,166 @@
+            utils.logger.error("There is no 'label' key in the provided glossary.")
+            return None
+        for lab_id, label in enumerate(glossary["labels"]):
+            if "aggregate" in config_filename:
+                self.add_label(lab_id, label["name"], label["color"],
+                               label["evaluate"], label["family"],
+                               label["contains_id"], label["contains"])
+            else:
+                name_items = label["name"].split('--')
+                self.add_label(lab_id, name_items[-1], label["color"],
+                               label["evaluate"], name_items[0])
+
+    def group_image_label(self, image):
+        """Group the labels
+
+        If the label ids 4, 5 and 6 belong to the same group, they will be turned
+        into the label id 4.
+
+        Parameters
+        ----------
+        image : PIL.Image
+
+        Returns
+        -------
+        PIL.Image
+        """
+        # turn all label ids into the lowest digits/label id according to its "group"
+        # (manually built)
+        a = np.array(image)
+        for root_id, label in enumerate(self.label_info):
+            for label_id in label['aggregate']:
+                mask = a == label_id
+                a[mask] = root_id
+        return Image.fromarray(a, mode=image.mode)
+
+    def _preprocess(self, image_filename, output_dir, aggregate, labelling=True):
+        """Resize/crop then save the training & label images
+
+        Parameters
+        ----------
+        datadir : str
+        image_filaname : str
+        aggregate : boolean
+        labelling : boolean
+
+        Returns
+        -------
+        dict
+            Key/values with the filenames and label ids
+        """
+        # open original images
+        img_in = Image.open(image_filename)
+
+        # resize images (self.image_size*larger_size or larger_size*self.image_size)
+        img_in = utils.resize_image(img_in, self.image_size)
+
+        # crop images to get self.image_size*self.image_size dimensions
+        crop_pix = np.random.randint(0, 1 + max(img_in.size) - self.image_size)
+        final_img_in = utils.mono_crop_image(img_in, crop_pix)
+
+        # save final image
+        new_in_filename = os.path.join(output_dir, 'images',
+                                       os.path.basename(image_filename))
+        final_img_in.save(new_in_filename)
+
+        # label_filename vs label image
+        if labelling:
+            label_filename = image_filename.replace("images/", "labels/")
+            label_filename = label_filename.replace(".jpg", ".png")
+            img_out = Image.open(label_filename)
+            img_out = utils.resize_image(img_out, self.image_size)
+            img_out = utils.mono_crop_image(img_out, crop_pix)
+            # group some labels
+            if aggregate:
+                img_out = self.group_image_label(img_out)
+
+            labels = utils.label_building(img_out,
```

Comment:
`label_building` is confusing with the notion of "building" in some other datasets. If it's the verb and not the noun, I prefer to use the infinitive form "build".

Reply:
Fixed, and renamed as `build_labels`.